lib/logmsg/logmsg-serialize.c

```diff
@@ -126,17 +166,323 @@ _deserialize_message(LogMessageSerializationState *state)
+  sd_element_root[logmsg_sd_prefix_len + sd_id_len]='.';
+
+
+  if (!log_msg_read_sd_param_first(sa, sd_element_root, self, &has_more_param))
+    goto error;
+
+  while (has_more_param)
+    {
+      if (!log_msg_read_sd_param(sa, sd_element_root, self, &has_more_param))
+        goto error;
+    }
+  g_free(sd_id);
+  *has_more = TRUE;
+  return TRUE;
+
+error:
+  g_free(sd_id);
+  return FALSE;
+
+}
+
+static gboolean
+log_msg_read_sd_data(LogMessage *self, SerializeArchive *sa)
+{
+  gboolean has_more_element = TRUE;
+  if (!log_msg_read_sd_element(sa, self, &has_more_element))
+    goto error;
+
+
+  while (has_more_element)
+    {
+      if (!log_msg_read_sd_element(sa, self, &has_more_element))
+        goto error;
+    }
+  return TRUE;
+error:
+
+  return FALSE;
+}
+
+static gboolean
+_deserialize_message_version_1x(LogMessageSerializationState *state)
+{
+  gsize stored_len;
+  gchar *source, *pid;
+  gchar *msgid;
+
+  LogMessage *msg = state->msg;
+  SerializeArchive *sa = state->sa;
+
+  if(state->version == LGM_V10)
+    {
+      guint16 stored_flags16;
+      if (!serialize_read_uint16(sa, &stored_flags16))
+        return FALSE;
+      msg->flags = stored_flags16;
+    }
+  else
+    {
+      if (!serialize_read_uint32(sa, &msg->flags))
+        return FALSE;
+    }
+  msg->flags |= LF_STATE_MASK;
+  if (!serialize_read_uint16(sa, &msg->pri))
+    return FALSE;
+
+  if (!serialize_read_cstring(sa, &source, &stored_len))
+    return FALSE;
+  log_msg_set_value(msg, LM_V_SOURCE, source, stored_len);
+  free(source);
+  if (!g_sockaddr_deserialize(sa, &msg->saddr))
+    return FALSE;
+
+  if (!timestamp_deserialize_legacy(sa, msg->timestamps))
+    return FALSE;
+
+  if(state->version == LGM_V12)
+    {
+      if (!log_msg_read_tags(msg, sa))
+        return FALSE;
+    }
+  if(!log_msg_read_common_values(msg, sa))
+    return FALSE;
+  if (!serialize_read_cstring(sa, &pid, &stored_len))
+    return FALSE;
+  log_msg_set_value(msg, LM_V_PID, pid, stored_len);
+  free(pid);
+
+  if (!serialize_read_cstring(sa, &msgid, &stored_len))
+    return FALSE;
+  log_msg_set_value(msg, LM_V_MSGID, msgid, stored_len);
+  free(msgid);
+  if (!serialize_read_uint8(sa, &msg->num_matches))
+    return FALSE;
+  if(!log_msg_read_matches_details(msg,sa))
+    return FALSE;
+  if (!log_msg_read_values(msg, sa) ||
+      !log_msg_read_sd_data(msg, sa))
+    return FALSE;
+  return TRUE;
+}
+
 static gboolean
 _check_msg_version(LogMessageSerializationState *state)
 {
   if (!serialize_read_uint8(state->sa, &state->version))
     return FALSE;
 
-  if (state->version != 26)
+  if (state->version < LGM_V10 || state->version > LGM_V26)
     {
-      msg_error("Error deserializing log message, unsupported version, "
-                "we only support v26 introduced in " VERSION_3_8 ", "
-                "earlier versions in syslog-ng Premium Editions are not supported",
+      msg_error("Error deserializing log message, unsupported version, ",
```

Comment:
The original code did not finish the string with comma. Is it needed?

Reply:
Nice catch :)